doc/charm++/manual.rst

```diff
@@ -2894,96 +2922,85 @@ elements migratable (see migration section above) and choose a load
-   prevent migration be performed. This is the automatic way of doing
-   load balancing where the application does not need to define
-   ResumeFromSync().
+   Note that ``AtSync()`` is not a blocking call, it just gives a hint
+   to load balancing that it is time for load balancing. The object
+   may be migrated during the time between ``AtSync()`` and
+   ``ResumeFromSync()``. One can choose to let objects continue
+   working with incoming messages; however, keep in mind the object may
+   suddenly show up in another processor, so make sure no operations
+   that could possibly prevent migration be performed. This is the
+   automatic way of doing load balancing where the application does
+   not need to define ``ResumeFromSync()``.
 
    The more commonly used approach is to force the object to be idle
-   until load balancing finishes. The user places an AtSync call at the
-   end of some iteration and when all elements reach that call load
-   balancing is triggered. The objects can start executing again when
-   ResumeFromSync() is called. In this case, the user redefines
-   ResumeFromSync() to trigger the next iteration of the application.
-   This manual way of using the at sync mode results in a barrier at
-   load balancing (see example here :numref:`lbexample`).
-
-#. **Manual mode**: The load balancer can be programmed to be started
-   manually. To switch to the manual mode, the application calls
-   *TurnManualLBOn()* on every processor to prevent the load balancer
-   from starting automatically. *TurnManualLBOn()* should be called as
-   early as possible in the program. It could be called at the
-   initialization part of the program, for example from a global
-   variable constructor, or in an initproc call
-   (Section :numref:`initproc`). It can also be called in the
-   constructor of a static array or before the *doneInserting* call for
-   a dynamic array. It can be called multiple times on one processor,
-   but only the last one takes effect.
-
-   The function call *CkStartLB()* starts load balancing immediately.
-   This call should be made at only one place on only one processor.
-   This function is not blocking, the object will continue to process
-   messages and the load balancing, when triggered, happens in the
-   background.
-
-   *TurnManualLBOff()* turns off manual load balancing and switches back
-   to the automatic load balancing mode.
+   until load balancing finishes. The user calls ``AtSync()`` at the
+   end of some iteration, then, when all elements reach that call,
+   load balancing is triggered. The objects can start executing again
+   when ``ResumeFromSync()`` is called. In this case, the user
+   redefines ``ResumeFromSync()`` to trigger the next iteration of the
+   application. This patter effectively results in a barrier at load
+   balancing time (see example here :numref:`lbexample`).
+
+#. **Periodic load balancing mode**: This mode uses a timer to perform
```

Comment:
[nitpick] the ordering of the overview and API details are reversed (above you introduce periodic before at-sync, and here you explain at-sync before periodic), so the guide might be nicer if the same ordering appeared for both.

Reply:
Good catch, I'll fix that.